presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -310,17 +310,18 @@ public boolean isNull(int position)
     }
 
     @Override
-    public Block getPositions(int[] positions)
+    public Block getPositions(int[] positions, int offset, int length)
     {
-        int length = positions.length;
+        checkValidPositions(positions, offset, length, positionCount);
+
         int[] newIds = new int[length];
         boolean isCompact = isCompact() && length >= dictionary.getPositionCount();
         boolean[] seen = null;
         if (isCompact) {
             seen = new boolean[dictionary.getPositionCount()];
         }
         for (int i = 0; i < length; i++) {
-            newIds[i] = getId(positions[i]);
+            newIds[i] = getId(positions[offset + i]);
```

Comment:
nit: `for (int i = offset; i < offset + length; ++i)`

Reply:
can't do it here as 0 indexed `i` is needed.